internal/storage/memory/memory.go

```diff
@@ -46,26 +49,75 @@ func (s *InMemStorage) PutURL(_ context.Context, r storage.URLRecord) error {
 	return nil
 }
 
-func (s *InMemStorage) PutBatchURLs(ctx context.Context, urls []storage.URLRecord) error {
+func (s *InMemStorage) SaveURLBatch(ctx context.Context, urls []*storage.URLRecord) error {
 	for _, r := range urls {
-		if err := s.PutURL(ctx, r); err != nil {
+		if err := s.SaveURL(ctx, r); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
-func (s *InMemStorage) GetURL(_ context.Context, id string) (*storage.URLRecord, error) {
+func (s *InMemStorage) GetURL(_ context.Context, shortURL string) (string, error) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	record, ok := s.db[id]
+	originalURL, ok := s.urls[shortURL]
 	if !ok {
-		return nil, fmt.Errorf("no url for id: %s", id)
+		return "", fmt.Errorf("no url for id: %s", shortURL)
 	}
 
-	return &record, nil
+	return originalURL, nil
+}
+
+func (s *InMemStorage) GetURLsByUserID(_ context.Context, userID string) ([]*storage.URLRecord, error) {
```

Comment:
```suggestion
func (s *InMemStorage) GetURLsByUserID(_ context.Context, userID string) ([]storage.URLRecord, error) {
```
Тоже лучше передавать без ссылки на структуру